docs/compile.md

```diff
@@ -355,3 +355,82 @@ Available overrides:
 - `WOLFBOOT_LIB_WOLFTPM`: Path to the [wolfTPM](https://github.com/wolfSSL/wolfTPM) library source code
 - `WOLFBOOT_LIB_WOLFPKCS11`: Path to the [wolfPKCS11](https://github.com/wolfssl/wolfpkcs11) library source code
 - `WOLFBOOT_LIB_WOLFHSM`: Path to the [wolfHSM](https://github.com/wolfSSL/wolfHSM) library source code
+
+## Key Generation and Signing
+
+### Default Key Behavior
+
+When building wolfBoot for the first time, the build system automatically generates the cryptographic keys needed for firmware signing and verification.
+
+- **Private Key**: `wolfboot_signing_private_key.der` - Used to sign firmware images
+- **Keystore**: `src/keystore.c` - Contains the public key embedded in the bootloader
+
+The key algorithm is determined by the `SIGN` variable (e.g., `SIGN=ECC256`, `SIGN=RSA2048`).
+
+For most targets, the makefile also builds the wolfBoot test app and signs it with the aforementioned key.
+
+### User-Provided Keys
+
```

Comment:
the title "User-Provided keys" is too generic. We already support importing public keys for third-party managed private keys.

Remind multiple repeated  "-i pubkey" and "-g privkey" are already supported, perhaps link back to "### Keygen tool" section above.

Expand on  why this is different (Importing both public + private keys that are generated and managed locally by the user,  vs. existing keygen -i in those cases where manufacturer or third party webservice manages the private keys)